src/sources/shopify/tracker/transform.js

```diff
@@ -242,83 +237,82 @@ const processEvent = async (inputEvent, metricMetadata) => {
     return NO_OPERATION_SUCCESS;
   }
 
+  let redisData;
   // Special handling for carts_update
   if (shopifyTopic === 'carts_update') {
     const result = await handleCartsUpdate(event, metricMetadata);
     if (result.invalidEvent) {
       return NO_OPERATION_SUCCESS;
     }
-    return enrichMessage(result.message, event, shopifyTopic, metricMetadata, result.redisData);
+    redisData = result.redisData;
```

Comment:
@koladilip we should update `redisData` before invalidEvent validation. Otherwise, next set of events doesn't get redisData & stitching doesn't happen

Reply:
let me check

Reply:
I meant `redisData` variable initialisation but as discussed, anonymousId stitching code block tries to fetch data from redis again if missing